src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxDocumentConverter.cs

```diff
@@ -0,0 +1,252 @@
+                    return;
+
+                case ValueType.Convertible:
+                    scope.Write(field.Id, ((IConvertible)value).ToString(CultureInfo.InvariantCulture), ref entryWriter);
+                    return;
+
+                case ValueType.Enumerable:
+                    var iterator = (IEnumerable)value;
+                    var enumerableScope = new EnumerableWriterScope(field.Id, ref entryWriter, scope.GetLengthList());
+                    foreach (var item in iterator)
+                    {
+                        InsertRegularField(field, item, indexContext, out _, ref entryWriter, enumerableScope);
+                    }
+                    enumerableScope.Finish(field.Id, ref entryWriter);
+                    return;
+
+                case ValueType.DynamicJsonObject:
+                    HandleObject((BlittableJsonReaderObject)value, field, indexContext, out _, ref entryWriter, scope);
+                    return;
+
+                case ValueType.ConvertToJson:
+                    var val = TypeConverter.ToBlittableSupportedType(value);
+                    if (!(val is DynamicJsonValue json))
+                        InsertRegularField(field, val, indexContext, out _, ref entryWriter, scope, nestedArray);
+                    return;
+
+                case ValueType.BlittableJsonObject:
+                    HandleObject((BlittableJsonReaderObject)value, field, indexContext, out _, ref entryWriter, scope);
+                    return; 
+
+                case ValueType.BoostedValue:
+                case ValueType.Stream:
+                case ValueType.DynamicNull:
+                case ValueType.Null:
+                default:
+                    throw new NotImplementedException();
+            }
+
+            shouldSkip = true;
+        }
+
+        void HandleArray(IEnumerable itemsToIndex, IndexField field, JsonOperationContext indexContext, out bool shouldSkip, ref IndexEntryWriter entryWriter, IWriterScope scope, bool nestedArray = false)
+        {
+            shouldSkip = false;
+            if (nestedArray)
+            {
+                return;
+            }
+
+            foreach (var itemToIndex in itemsToIndex)
+            {
+                InsertRegularField(field, itemToIndex, indexContext, out _, ref entryWriter, scope);
+            }
+        }
+
+        //todo maciej Discuss how we gonna handle nestedArrays. Now I skip them.
+        void HandleObject(BlittableJsonReaderObject val, IndexField field, JsonOperationContext indexContext, out bool shouldSkip, ref IndexEntryWriter entryWriter, IWriterScope scope, bool nestedArray = false)
+        {
+            if (val.TryGetMember(Constants.Json.Fields.Values, out var values) &&
+                IsArrayOfTypeValueObject(val))
+            {
+                HandleArray((IEnumerable)values, field, indexContext, out _, ref entryWriter, scope, true);
+            }
+
+            shouldSkip = false;
```

Comment:
Question here is what we want to do when user is indexing an object. In Lucene we save the value as JSON, but I'm wondering if this make sense to carry this behavior to Corax.
The good thing here is that this allows us to do a "complex" object and compare against that.

Not sure what actual use we are seeing here, and use can at worst force a string serialization if needed

Reply:
@ayende What could be another approach to deal with an object here? What were you thinking about?

Reply:
I think we can say that we'll skip the object for now.
If the user wants that, they can force it to be a string.

Reply:
Note that in auto indexes users can't just force it, but still they might query against complex objects (even if that's rare case). My concern is that this will be breaking change for some cases (not very popular ones but still an isse).

What is the potential benefit of dropping the current behavior in Corax? My thinking is that it was working flawlessly so far so what are we trying to gain here by not supporting that. Note that this doesn't affect us in most cases anyway because indexing an object isn't something we see as common practice.

Reply:
I will add a to-do in code to not forget about it.

Reply:
Primary reason is that this gives us a _chance_ to revert this decision, with a clear boundary on backward compact maybe broken.
Mostly I would like to do that because it usually doesn't make sense, we also had some cases where the object was literally MB in size.

Reply:
Not critical either way to me.

Reply:
I'm worried that we might loose more by dropping this than gain any value. I agree that having few MBs docs is an issue but I'm in favor of exposing that warning to a user rather than blocking this option completely (similar warning like with huge documents). Anyway, as you already said they could still force string serialization.

@ppekrol your thoughts on this?

Reply:
I think it would be better to handle this to cover as much as we can functionality of Lucene in Corax.